lib/sshkit.ex

```diff
@@ -90,13 +99,26 @@ defmodule SSHKit do
   hosts = ["10.0.0.1", "10.0.0.2"]
   context = SSHKit.context(hosts)
   ```
+
+  Any shared options can be specified in the second argument.
+  Here we add a user and port for all hosts.
+
+  ```
+  hosts = ["10.0.0.1", "10.0.0.2"]
+  options = [user: "admin", port: "2222"]
+  context = SSHKit.context(hosts, options)
+  ```
   """
-  def context(hosts) do
+
+  def context(hosts, options \\ []) do
     hosts =
       hosts
       |> List.wrap
-      |> Enum.map(&host/1)
-    %Context{hosts: hosts}
+      |> Enum.map(
+        fn(h) ->
+          host(h, options)
+        end)
```

Comment:
for easier readability you could extract the `Enum.map` part into a function:

```elixir
defp build_host(hosts, shared_options) do
  build_host = &host(&1, shared_options)
  Enum.map(hosts, build_host)
end
```

so that `context` reads like this:

```elixir
def context(hosts, shared_options \\ []) do
  hosts =
    hosts
    |> List.wrap
    |> build_hosts(shared_options)

  %Context{hosts: hosts}
end
```